### DISCUSSION TITLE
Glowroot and opentelemetry dependencies in same time

### DISCUSSION THREAD

**Discussion by PandeoF1** [open-telemetry/opentelemetry-java-instrumentation#11082] [Q&A] [answered]
Hi, do i will have some problem to use the Java Agent Glowroot and the dependencies in same time ?

**Comment by steverao** [comment]
Sorry, I am not familiar with [Glowroot](https://glowroot.org/features.html) so much, But from relevant document, it provides similar abilities with OTel. Why you want to use them in same time? 
By the way, I found @trask you are the main contributors of [Glowroot](https://github.com/glowroot/glowroot) project：）Do you have any comment about the discussion?

**Comment by PandeoF1** [comment]
> Sorry, I am not familiar with [Glowroot](https://glowroot.org/features.html) so much, But from relevant document, it provides similar abilities with OTel. Why you want to use them in same time? By the way, I found @trask you are the main contributors of [Glowroot](https://github.com/glowroot/glowroot) project：）Do you have any comment about the discussion?

Actually we need Glowroot to have a small interface with the agglomeration of trace without any other tools, so my goal is to have Glowroot for the default deployment and if the we want to deploy Otel to have our collector, jaeger...

But if we can have the both in same time and just deactivate Otel with a env var it will be perfect !

**Reply by trask** [reply]
hi @PandeoF1!

it's certainly worth trying. I would try both ways, with `-javaagent` arg for glowroot first and opentelemetry second, and vice versa.

check out #1534 for the project's official stance on running alongside other monitoring java agents (basically it's best effort only, we do not make any guarantees)

**Reply by PandeoF1** [reply]
:wave: !

> it's certainly worth trying. I would try both ways, with -javaagent arg for glowroot first and opentelemetry second, and vice versa.

I will try that tomorrow and tell you if their is a diff between OTEL with and without glowroot !

> check out https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1534 for the project's official stance on running alongside other monitoring java agents (basically it's best effort only, we do not make any guarantees)

That makes sense! Our use case is really specific ;)

**Reply by PandeoF1** [reply, marked as answer]
Hello !
So after some day i have made a test with opentelemetry dependencies and glowroot agent. (and the same test without opentelemetry and another one without glowroot...)
So i didn't saw a difference between both traces / metrics when the both agent are set or not.
The only diff was the precision of the trace with the dependencies because i haven't use all the dep that i needed (spring boot for example..).

So in fact the only diff is the cpu and memory consumption when the both agent are in the same time !

IMPORTANT: I have check only the diff on the side of opentelemetry. On the webui of glowroot i have only check if the data are retrieved and not if they are the same between all test (I dont' have the time actually)
